pkg/controllers/addonmanagement/managedcluster.go

```diff
@@ -326,18 +377,31 @@ func (c *clusterController) applyWorkloadCluster(ctx context.Context, workspaceH
 			"apiVersion": "workload.kcp.dev/v1alpha1",
 			"kind":       "WorkloadCluster",
 			"metadata": map[string]interface{}{
-				"name": clusterName,
+				"name":   clusterName,
+				"labels": labels,
 			},
 			"spec": map[string]interface{}{
 				"kubeconfig": "",
 			},
 		},
 	}
-	if _, err := dynamicClient.Resource(clusterGVR).Create(ctx, workloadCluster, metav1.CreateOptions{}); err != nil {
-		return err
-	}
 
-	c.eventRecorder.Eventf("KCPWorkloadClusterCreated", "The KCP workload cluster %s is created in workspace %s", clusterName, workspaceConfig.Host)
+	if wc == nil {
+		if _, err := dynamicClient.Resource(clusterGVR).Create(ctx, workloadCluster, metav1.CreateOptions{}); err != nil {
+			return err
+		}
+
+		c.eventRecorder.Eventf("KCPWorkloadClusterCreated", "The KCP workload cluster %s is created in workspace %s", clusterName, workspaceConfig.Host)
+	} else {
+		// TODO - Does this need to be a PATCH? (Probably.) Are we going to be in conflict with updates being made on the object by KCP? Assume
+		// we control the spec and KCP controls the status.
```

Comment:
yeah, patch is good choice to avoid conflict, I think we can make a simple implement firstly :)